### ISSUE TITLE
Wizard to allow 'steps'

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-docs#3226]
One of the goal of the wizard is to present the minimum information required to get a user started.

As Sentry adds more features and more integrations, exposing the user to such things in the wizard can be in conflict with this principle.

Take for example the React Native wizard:

https://github.com/getsentry/sentry-docs/blob/c61af7ca914ec343ab4c451985bfa636d2e1aecd/src/wizard/react-native/index.md

It's a short as it can be to get crashes and release health (which is enabled by default) working.

At the end of it, we could offer a Next step (optional):

> Want to measure performance? Click **Next**.

Then provide the user with options:

1. Using React Navigation v4 router?
   * Show the RN router v4 instructions
2. Using React Navigation v5 router?
   * Show the RN router v5 instructions
3. Using a different router? 
  * Show the "manual" instrumentation as an example to start/finish transactions
  * Would be nice to get feedback so we can learn from users

Other use cases include:
Spring, Spring Boot, ASP.NET, ASP.NET Core all are often used in tandem with a logging integration.
The wizard could present the user the option to chose a logging library (we support many) and present the instructions.

Android users often use Timber as a logging library. We're working on `OkHttp` integration for HTTP requests. 

Bottom line is that with the current architecture of a single logo/title/page we're limited in how we can guide the users through the best setup for them, and also learn from potential integrations they are looking and we don't support.

Relates to #2719

**Comment by PeloWriter** [comment]
Yes, I was just discussing this with @jeremybourque. We want folks to get up and running quickly, but we need to balance this with ensuring we're providing the correct information (if you're using JS, are you using React; if you're using Java, are you using SpringBoot, and so forth). 

A smarter wizard would ensure users are getting to the right content.

In addition, a smarter wizard would automatically reflect changes made to the code samples in the core docs (would also be useful if these changes were automatically reflected in our marketing sites).

**Comment by bruno-garcia** [comment]
Some other use cases:

## Ionic wizard

Select a JS framework:
* Angular
* Vue
* React

Select a runtime:
* Capacitor
* Cordova


## React Native

* Basic install steps
* (required for performance monitoring) Select a routing integration
  * React Navigation 4
  * React Navigation 4
  * ...

Then we can show appropriate install instructions for performance montoring for those libraries

## .NET

This could be a façade wizard that calls to another depending on user options:

Select a framework
* ASP.NET (could invoke another wizard passing the language which was alre
* WPF
* WinForms
* Xamarin
* Other: Show raw C#/F# example (can we get input from the user to help us priortize new integrations?)

Select a language:
* F#
* C#

Add a logging library?
* Serilog
* NLog
* log4net

## Spring

Select a language
* Kotlin
* Java

It's not just a code snippet change, if Kotlin: Additional package and steps to get Coroutines working

Framework
* Spring
* Spring Boot
* Servlet

Logging
* logback
* log4j2

## Android

Select language
* Java
* Kotlin

If Kotlin, additional info (see above)

Using OkHttp? - Steps to install and get spans
Using Timber? -> Steps to install and get crumbs and error events

**Comment by bruno-garcia** [comment]
Relates to: https://github.com/getsentry/sentry-docs/issues/2719

Another reason this came up: On Android we want to prioritize installing with the Android Gradle Plugin so we can automate installing extra packages and configure them via bytecode manipulations when needed.
The problem is that a non trivial amount of users won't be able to leverage this because they are still stuck on a version of AGP we don't support. Ideally we'd ask the user first: "Do you use AGP v7 or higher?" and lead them to the right install steps.

**Comment by Jesse-Box** [comment]
@bruno-garcia 

Since, the telemetry experience has started a [major refactor](https://github.com/getsentry/sentry/issues/48144) as to how our wizards are being generated. A few SDK teams are already asking how this refactor can benefit the end-user experience because it allows us to render instructions specific to your language, framework, package manager etc.

Here are some example tickets that are on our radar.
1. https://github.com/getsentry/sentry/issues/50226
2. https://github.com/getsentry/sentry/issues/51586

cc: @matejminar @ale-cota
